src/execution_plan.c

```diff
@@ -327,37 +325,40 @@ static void ExecutionPlan_Distribute(ExecutionPlan* ep){
 
 static Record* ExecutionPlan_FilterNextRecord(ExecutionPlan* ep, ExecutionStep* step, RedisModuleCtx* rctx){
     Record* record = NULL;
+    INIT_TIMER;
     while((record = ExecutionPlan_NextRecord(ep, step->prev, rctx))){
         if(record == &StopRecord){
             return record;
         }
         if(RedisGears_RecordGetType(record) == ERROR_RECORD){
             return record;
         }
+	    START_TIMER;
         char* err = NULL;
         bool filterRes = step->filter.filter(rctx, record, step->filter.stepArg.stepArg, &err);
         if(err){
             ep->isErrorOccure = true;
             RedisGears_FreeRecord(record);
             record = RG_ErrorRecordCreate(err, strlen(err) + 1);
-            return record;
+            goto end;
         }
         if(filterRes){
-            return record;
+            goto end;
         }else{
             RedisGears_FreeRecord(record);
         }
     }
-    return NULL;
+    return NULL;  // TODO: check if this is a reachable code path
```

Comment:
its reachable when there is no more records

Reply:
Ok, cool.

Reply:
Resolved via  f227215